### PR TITLE
Add support for `f32`, as well as `f64`

### PR DESCRIPTION
**Pull request by smu160**
This PR should address issue #14

### CODE REVIEW

**Comment by Shnatsel** [comment]
It seems like it fails to build on CI with a syntax error?

**Comment by smu160** [comment]
> It seems like it fails to build on CI with a syntax error?

Yes. Sorry, I should have converted this to a draft. I have local changes almost ready to check-in. They should address the aforementioned issue, as well as others.

File: src/kernels.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ macro_rules! fft_butterfly_n_simd {
 }
 
 fft_butterfly_n_simd!(fft_64_chunk_n_simd, f64, 8, f64x8);
-fft_butterfly_n_simd!(fft_32_chunk_n_simd, f32, 8, f32x8);
```
Comment (smu160) [inline src/kernels.rs]: @Shnatsel Bumped up number of lanes to 16 when using `AVX-512` and `f32`

File: src/kernels.rs [hunk 2]
```diff
@@ -49,7 +49,7 @@ macro_rules! fft_butterfly_n_simd {
...
+fft_butterfly_n_simd!(fft_32_chunk_n_simd, f32, 16, f32x16);
```
Comment (smu160) [inline src/kernels.rs:52]: @Shnatsel Bumped up # of lanes to 16 when using `AVX-512` and `f32`

**Review by smu160 — COMMENTED** [review]
@Shnatsel I think this should be good to go now

**Pull request merged**